apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/sink/direct/AbstractClaimRdaSink.java

```diff
@@ -142,15 +146,36 @@ public void updateLastSequenceNumber(long lastSequenceNumber) {
    * @param apiVersion The version of the api used to get the message.
    * @param message The message that was being transformed when the error occurred.
    * @param exception The exception that was thrown while transforming the message.
-   * @throws IOException If there was an issue writing to the database.
+   * @throws IOException if there was an issue writing to the database.
+   * @throws ProcessingException If the error limit was reached.
    */
   @Override
   public void writeError(
       String apiVersion, TMessage message, DataTransformer.TransformationException exception)
-      throws IOException {
+      throws IOException, ProcessingException {
     entityManager.getTransaction().begin();
     entityManager.merge(createMessageError(apiVersion, message, exception.getErrors()));
     entityManager.getTransaction().commit();
+
+    checkErrorCount();
+  }
+
+  /**
+   * Checks if the allowed number of unresolved {@link MessageError}s in the database was exceeded.
+   *
+   * @throws ProcessingException If the error limit was reached.
+   */
+  @Override
+  public void checkErrorCount() throws ProcessingException {
+    var query =
+        entityManager.createQuery(
+            "select error from MessageError error where status = :status", MessageError.class);
+    query.setParameter("status", MessageError.Status.UNRESOLVED);
+    int errorCount = query.getResultList().size();
+
+    if (errorCount > errorLimit) {
+      throw new ProcessingException(new IllegalStateException("Error limit reached"), 0);
```

Comment:
silly nit-pick...
do we wanna consider when we've reached the limit rather than past it ?
(errorCount >= errorLimit)

Reply:
My definition of `errorLimit` here is that it's the maximum allowed number of errors.  If you were to set `errorLimit` to `0`, it would mean "Stop running if there are any errors".  Using `errorCount >= errorLimit` would cause it to fail even when there are no errors in the database.